examples/pytorch/preprocessing_dist_training/metis_creation.py

```diff
@@ -0,0 +1,89 @@
+import pandas as pd
+import os
+import glob
+import json
+import argparse
+from collections import defaultdict
+
+path = os.getcwd()
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-n", "--name", help="name of graph to create", default="order")
+parser.add_argument("-nc", "--node_column", nargs="+", default=['order_id', 'entity_index', 'order_datetime', 'cid'])
+parser.add_argument("-nk", "--node_key", default='entity_index')
+parser.add_argument("-ec", "--edge_column", nargs="+", default=['predicate_type', 'predicate_index', 'entity_index', 'entity_index_y'])
+parser.add_argument("-es", "--edge_start", default="entity_index")
+parser.add_argument("-en", "--edge_end", default="entity_index_y")
+args = parser.parse_args()
+
+
+#Store all types of node in nodes folder
+nodes_list = sorted(glob.glob(os.path.join(path, "nodes/*")))
+
+if os.path.exists("{}_nodes.txt".format(args.name)):
+    os.remove("{}_nodes.txt".format(args.name))
+
+schema_dict = defaultdict(dict)
+
+node_type_id = 0
+all_nodes_count = 0
+for node_type_name in nodes_list:
+    nodes_count = 0
+    csv_files = sorted(glob.glob(os.path.join(node_type_name, "*.csv")))
+    for file_name in csv_files:
+        df = pd.read_csv(file_name, error_bad_lines=False, escapechar='\\', names=args.node_column, usecols=[*range(len(args.node_column))])
```

Comment:
can you benchmark it on a large graph? my experience is that pandas is slow in reading a large csv file. you might want to numpy (https://docs.scipy.org/doc/numpy-1.9.2/reference/generated/numpy.genfromtxt.html). or you can use awk to get the required column first and then load it to python.

Reply:
this benchmark was done on 1.5G Data.


ubuntu@ip-10-0-27-95:~/jul12Final$ time python3 benchmarknumpy.py 
real	1m26.509s
user	1m22.064s
sys	0m5.973s


ubuntu@ip-10-0-27-95:~/jul12Final$ time python3 benchmarkpandas.py 
real	0m12.470s
user	0m11.452s
sys	0m2.770s


Here it shows pandas is faster than numpy.